strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java

```diff
@@ -0,0 +1,29 @@
+package org.carlspring.strongbox.authentication.api.impl.ldap;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class LdapAuthenticator
```

Comment:
> @fuss86 So I think you can put a AuthenticationProvider synchronized collection there and just manage that collection.

Yes, it's possible to do it that way. Bear in mind that this collection needs to have fully synchronized replaceAll atomic (in the synchronization meaning) operation like [AuthenticatorsRegistry](https://github.com/strongbox/strongbox/blob/master/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java) already has (+ atomic `reorder` method). Of course it's doable but requires re-work here.